src/database/auth.rs

```diff
@@ -3,94 +3,104 @@ use argon2::{
-            Ok(user_identities
-                .filter(lower(username).eq(target_username.to_lowercase()))
-                .first::<UserIdentity>(&mut conn)?)
-        })
-        .await
+        Ok(user_identities
+            .filter(lower(username).eq(target_username.to_lowercase()))
+            .first::<UserIdentity>(&mut conn)
+            .await?)
     }
 
     pub async fn delete_user(&self, userid: i32) -> Result<bool, TimeError> {
+        let mut conn = self.db.get().await?;
         use crate::schema::user_identities::dsl::*;
 
-        self.run_async_query(move |mut conn| {
-            Ok(diesel::delete(user_identities.filter(id.eq(userid))).execute(&mut conn)? > 0)
-        })
-        .await
+        Ok(diesel::delete(user_identities.find(userid))
+            .execute(&mut conn)
+            .await?
+            > 0)
     }
 
     pub async fn get_user_by_id(&self, userid: i32) -> Result<UserIdentity, TimeError> {
+        let mut conn = self.db.get().await?;
         use crate::schema::user_identities::dsl::*;
 
-        self.run_async_query(move |mut conn| {
-            Ok(user_identities
-                .filter(id.eq(userid))
-                .first::<UserIdentity>(&mut conn)?)
-        })
-        .await
+        Ok(user_identities
+            .find(userid)
+            .first::<UserIdentity>(&mut conn)
+            .await?)
     }
 
+    // TODO: get rid of unwraps
     pub async fn verify_user_password(
         &self,
-        username: &str,
+        arg_username: &str,
         password: &str,
     ) -> Result<Option<UserIdentity>, TimeError> {
-        let user = self.get_user_by_name(username.to_string()).await?;
-        let tuser = self.get_testaustime_user_by_id(user.id).await?;
+        let mut conn = self.db.get().await?;
+
+        use user_identities::dsl::username;
+
+        let (user, tuser) = user_identities::table
+            .filter(username.eq(arg_username))
+            .inner_join(testaustime_users::table)
+            .first::<(UserIdentity, TestaustimeUser)>(&mut conn)
+            .await?;
 
         let argon2 = Argon2::default();
-        let Ok(salt) = SaltString::new(std::str::from_utf8(&tuser.salt).unwrap()) else {
+        let Ok(salt) = SaltString::new(std::str::from_utf8(&tuser.salt).expect("bug: impossible"))
+        else {
             return Ok(None); // The user has no password
         };
         let password_hash = argon2.hash_password(password.as_bytes(), &salt).unwrap();
-        if password_hash.hash.unwrap().as_bytes() == tuser.password {
+        if password_hash.hash.expect("bug: impossible").as_bytes() == tuser.password {
             Ok(Some(user))
         } else {
             Ok(None)
         }
     }
 
     pub async fn regenerate_token(&self, userid: i32) -> Result<String, TimeError> {
+        let mut conn = self.db.get().await?;
+
         let token = crate::utils::generate_token();
 
         let token_clone = token.clone();
```

Comment:
I'm no expert in Rust so feel free to disregard if this is invalid, but is this clone necessary anymore now that we don't have the `move |mut conn| { ... }`?

Reply:
You're right it's not required.